Shared/Managers/KeyboardManager.swift

```diff
@@ -37,88 +41,110 @@ class KeyboardManager {
         case none // No event has happened
     }
 
-    func reset() {
-        self.cachedKeyboardEndFrame = .zero
-        self.inputAccessoryView = nil
-        self.willKeyboardShow = false
-        self.isKeyboardShowing = false
-        self.cancellables.forEach { cancellable in
-            cancellable.cancel()
-        }
-        self.cancellables.removeAll()
+    static let shared = KeyboardManager()
+
+    private var cancellables = Set<AnyCancellable>()
+
+    @Published var currentEvent: KeyboardEvent = .none
+    @Published var cachedKeyboardEndFrame: CGRect = .zero
+    var isKeyboardShowing: Bool = false
+
+    init() {
+        self.addKeyboardObservers()
     }
 
-    func addKeyboardObservers(with inputAccessoryView: UIView? = nil) {
-        self.inputAccessoryView = inputAccessoryView
+    private func addKeyboardObservers() {
 
         NotificationCenter.default.publisher(for: UIResponder.keyboardWillShowNotification)
             .mainSink { (notification) in
-                self.currentEvent = .willShow(notification)
-                self.cachedKeyboardEndFrame = self.getFrame(for: notification)
-
-                if let inputView = self.inputAccessoryView {
-                    let willShow = self.cachedKeyboardEndFrame.height > inputView.height
-                    self.willKeyboardShow = willShow
-                } else {
-                    self.willKeyboardShow = true
+                guard !self.isKeyboardShowing else { return }
```

Comment:
The notification center sends a lot of redundant keyboard events for some reason.